lib/jxl/dec_external_image.cc

```diff
@@ -315,12 +315,12 @@ Status ConvertChannelsToExternal(const ImageF* channels[], size_t num_channels,
   // First channel may not be nullptr.
   size_t xsize = channels[0]->xsize();
   size_t ysize = channels[0]->ysize();
-
   if (stride < bytes_per_pixel * xsize) {
     return JXL_FAILURE("stride is smaller than scanline width in bytes: %" PRIuS
                        " vs %" PRIuS,
                        stride, bytes_per_pixel * xsize);
   }
+  JXL_ASSERT(out_callback || out_size >= ysize * stride);
```

Comment:
Not sure how I feel about adding an assert here, shouldn't we return a failure instead?

Reply:
Makes sense. Changed the condition to correspond to https://github.com/libjxl/libjxl/issues/994 (don't require padding after last row), and returning failure instead of having an assert.